Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/profile/ProfileCensor.kt

```diff
@@ -0,0 +1,102 @@
+package de.rki.coronawarnapp.bugreporting.censors.profile
+
+import dagger.Reusable
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.CensorContainer
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidAddress
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidCity
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidName
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidPhoneNumber
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidZipCode
+import de.rki.coronawarnapp.bugreporting.debuglog.internal.DebuggerScope
+import de.rki.coronawarnapp.profile.storage.ProfileRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import org.joda.time.format.DateTimeFormat
+import javax.inject.Inject
+
+@Reusable
+class ProfileCensor @Inject constructor(
```

Comment:
Is it really necessary to save all that stuff in here?
checkLog is suspending, so it would be possible to get the values from the flow directly